docs/cli/commands.md

```diff
@@ -70,6 +70,29 @@ Slash commands provide meta-level control over the CLI itself.
 - **`/stats`**
   - **Description:** Display detailed statistics for the current Gemini CLI session, including token usage, cached token savings (when available), and session duration. Note: Cached token information is only displayed when cached tokens are being used, which occurs with API key authentication but not with OAuth authentication at this time.
 
+- **`/terminal-setup`**
+  - **Description:** Automatically detects and configures your terminal emulator to work with gemini-cli.
+  - **Action:**
+    - Detects your current terminal emulator
+    - For supported terminals, automatically adds the necessary keybinding configuration
+    - Creates backups of configuration files before modifying them
+    - Reports success or failure with appropriate instructions
+  - **Supported Terminals:**
+    - **VS Code** - Adds keybindings to VS Code's `keybindings.json`
+    - **Cursor** - Adds keybindings to Cursor's `keybindings.json`
+    - **Windsurf** - Adds keybindings to Windsurf's `keybindings.json`
+  - **Terminals that work out of the box (no setup required):**
+    - **Alacritty**
+    - **Ghostty**
+    - **Foot**
+    - **iTerm2**
+    - **Kitty**
+    - **Rio**
+    - **WezTerm**
+  - **Notes:**
+    - Most terminals require a restart after configuration
+    - VS Code requires restarting the entire application, not just the terminal
```

Comment:
on Mac it seems like keybindings work out of the box in VSCode even without running the terminal-setup. I suspect some of your other fixes make it work by default. I verified and my `keybindings.json` was empty. Without your change I do not get proper shift-enter behavior on mac.

Reply:
Great! - think we are hitching a ride on the \\\r\n train probably.

Reply:
I am unable to replicate this on my mac machine. I need to do terminal-setup for shift-enter to work even with latest changes.